x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_embeddable_factory.tsx

```diff
@@ -0,0 +1,438 @@
+  // - `mlFieldFormatService` is a stateful legacy service that relied on "dependency cache",
+  //   so because of its own state it needs to be made available as a global service.
+  //   In the long run we should again try to get rid of it here and make it available via
+  //   its own context or possibly without having a singleton like state at all, since the
+  //   way this manages its own state right now doesn't consider React component lifecycles.
+  const mlIndexUtils = indexServiceFactory(pluginsStart.data.dataViews);
+  const mlFieldFormatService = fieldFormatServiceFactory(mlApiServices, mlIndexUtils);
+
+  return [
+    coreStart,
+    pluginsStart as MlDependencies,
+    {
+      anomalyDetectorService,
+      anomalyExplorerService,
+      mlApiServices,
+      mlCapabilities,
+      mlFieldFormatService,
+      mlJobService,
+      mlResultsService,
+      mlTimeSeriesSearchService,
+      mlTimeSeriesExplorerService,
+      toastNotificationService,
+    },
+  ];
+};
+
+export const getSingleMetricViewerEmbeddableFactory = (
+  getStartServices: StartServicesAccessor<MlStartDependencies, MlPluginStart>
+) => {
+  const factory: ReactEmbeddableFactory<
+    SingleMetricViewerEmbeddableState,
+    SingleMetricViewerEmbeddableApi
+  > = {
+    type: ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE,
+    deserializeState: (state) => {
+      return state.rawState as SingleMetricViewerEmbeddableState;
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const services = await getServices(getStartServices);
+      const subscriptions = new Subscription();
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+
+      const {
+        api: timeRangeApi,
+        comparators: timeRangeComparators,
+        serialize: serializeTimeRange,
+      } = initializeTimeRange(state);
+
+      const {
+        singleMetricViewerControlsApi,
+        serializeSingleMetricViewerState,
+        singleMetricViewerComparators,
+        onSingleMetricViewerDestroy,
+      } = initializeSingleMetricViewerControls(state, titlesApi);
+
+      const dataLoading = new BehaviorSubject<boolean | undefined>(true);
+      const blockingError = new BehaviorSubject<Error | undefined>(undefined);
+      const query$ =
+        // @ts-ignore property does not exist on type 'PresentationContainer'
+        (state.query ? new BehaviorSubject(state.query) : parentApi?.query$) ??
+        new BehaviorSubject(undefined);
+      const filters$ =
+        // @ts-ignore property does not exist on type 'PresentationContainer'
+        (state.query ? new BehaviorSubject(state.filters) : parentApi?.filters$) ??
+        new BehaviorSubject(undefined);
+
+      const refresh$ = new BehaviorSubject<void>(undefined);
+
+      const api = buildApi(
+        {
+          ...titlesApi,
+          ...timeRangeApi,
+          ...singleMetricViewerControlsApi,
+          query$,
```

Comment:
I see that the base ML api says that you must publish queries and filters because of `PublishesUnifiedSearch`. But is there anywhere that actually uses this information? It seems to me that you might not need filters or query on your own API, and could maybe just use the parent's.

Reply:
yes it can be used indeed. It just wasn't ready when I was working on the anomaly swim lane